Requirements/DesignPatterns.md

```diff
@@ -0,0 +1,129 @@
+## COSC 310 Two Possible Design Patterns for Project M3
+
+For our project we’re choosing the Model-View-Controller (MVC) design pattern as it’s the best for desktop and web applications. Some benefits of MVC specifically for our project (ticketing platform) include separation of concerns, flexibility, maintainability, code reusability, support for multiple platforms.
+
+Separation of concerns includes:
+
+- The model manages data and business logic
+- The view handles the user interface and presentation
+- The controller handles user input, processes requests, and manages flow
+
+Flexibility:
+
+- Each component (Model, View, Controller) can be developed independently, promoting modular and scalable code. Flexibility provides extensible architecture that can adapt to changing requirements and business needs.
+
+Maintainability:
+
+- Changes to one component do not affect the others, making it easier to maintain and update the system
+
+Code reusability:
+
+- Reusable components for user authentication, event management, and transaction processing can be implemented across the application
+
+Performance:
+
+- The MVC design pattern allows for efficient handling of user actions, ensuring that the system responds to use actions within the specified timeframe.
+
+Implementing the MVC design pattern for the Ticket purchasing platform adheres to industry standards, promotes code maintainability, and ensures a balance between reason (logical structure) and passion (user interface and experience)
+
+For our ticketing platform, TicketHub, our Model would consist of User, Event, and Transaction as shown below:
+
+1.  Model:
+
+    - a.) User Model:
+
+      - Manages user information, login status, and user roles (guest, buyer, seller, administrator).
+      - Handles user account creation, modification, and profile updates.
+      - limits user access based on their user type and login status
+
+    - b.) Event Model:
+
+      - Manages events, tickets, and related statistics.
+      - Handles the creation, modification, and review of events by sellers.
+
+    - c.) Transaction Model:
+      - Manages transactions, including ticket purchases and refunds.
```

Comment:
We are not including refunds in our model at this point in time so probably remove this. Instead, you could be more specific about the buyer and seller transaction mentioning that the transaction model will handle the funds transitioning from the buyer's specified payment option and into the seller's account.